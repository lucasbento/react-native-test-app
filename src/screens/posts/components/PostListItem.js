import React from 'react';
import PropTypes from 'prop-types';
import {TouchableOpacity, Text, Image, StyleSheet} from 'react-native';
import ArrowRight from '../../../assets/arrow-right.png';

const PostListItem = ({title, onPress}) => (
  <TouchableOpacity onPress={onPress} style={styles.container}>
    <Text style={styles.title}>{title}</Text>

    <Image source={ArrowRight} style={styles.arrowRightIcon} />
  </TouchableOpacity>
);

PostListItem.propTypes = {
  title: PropTypes.string.isRequired,
  onPress: PropTypes.func.isRequired,
};

const styles = StyleSheet.create({
  container: {
    margin: 5,
    padding: 10,
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center',
    backgroundColor: '#FFF',
    borderRadius: 5,
    shadowColor: '#000',
    shadowOffset: {
      width: 0,
      height: 1,
    },
    shadowOpacity: 0.18,
    shadowRadius: 1.0,
    elevation: 1,
  },
  title: {
    fontWeight: 'bold',
    marginRight: 15,
  },
  arrowRightIcon: {
    width: 10,
    height: 20,
    resizeMode: 'contain',
  },
});

export default PostListItem;
