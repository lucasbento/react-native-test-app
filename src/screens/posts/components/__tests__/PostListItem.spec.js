import React from 'react';
import {
  render,
  shallow,
  waitForElement,
  fireEvent,
} from '../../../../configuration/renderer';
import PostListItem from '../PostListItem';
import post from '../../../../../__fixtures__/post';

describe('<PostListItem />', () => {
  beforeEach(() => {
    this.props = {
      title: post.title,
      onPress: jest.fn(),
    };
  });

  it('renders correctly', async () => {
    const {output} = shallow(<PostListItem {...this.props} />);

    expect(output).toMatchSnapshot();
  });

  it('renders post title', async () => {
    const {getByText} = render(<PostListItem {...this.props} />);

    const title = await waitForElement(() => getByText(this.props.title));

    expect(title).toBeDefined();
  });

  it('calls `onPress` callback when pressed', async () => {
    const {getByText} = render(<PostListItem {...this.props} />);

    await waitForElement(() => fireEvent.press(getByText(this.props.title)));

    expect(this.props.onPress).toHaveBeenCalled();
    expect(this.props.onPress).toHaveBeenCalledTimes(1);
  });
});
