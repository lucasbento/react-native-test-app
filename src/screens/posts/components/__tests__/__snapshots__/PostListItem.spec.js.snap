// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PostListItem /> renders correctly 1`] = `
<TouchableOpacity
  activeOpacity={0.2}
  onPress={[MockFunction]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#FFF",
      "borderRadius": 5,
      "elevation": 1,
      "flexDirection": "row",
      "justifyContent": "space-between",
      "margin": 5,
      "padding": 10,
      "shadowColor": "#000",
      "shadowOffset": Object {
        "height": 1,
        "width": 0,
      },
      "shadowOpacity": 0.18,
      "shadowRadius": 1,
    }
  }
>
  <Text
    style={
      Object {
        "fontWeight": "bold",
        "marginRight": 15,
      }
    }
  >
    sunt aut facere repellat provident occaecati excepturi optio reprehenderit
  </Text>
  <Image
    source={
      Object {
        "testUri": "../../../src/assets/arrow-right.png",
      }
    }
    style={
      Object {
        "height": 20,
        "resizeMode": "contain",
        "width": 10,
      }
    }
  />
</TouchableOpacity>
`;
